commands/addons.js

```diff
@@ -259,32 +258,74 @@ module.exports = {
     help:        `The default filter applied depends on whether you are in a Heroku app
 directory. If so, the --app flag is implied. If not, the default of --all
 is implied. Explicitly providing either flag overrides the default
-behaviour.
+behavior.
 
 Examples:
 
   $ heroku ${topic} --all
-  $ heroku ${topic} --app acme-inc-website
-
-Identifying specific add-ons:
-
-  Add-ons have canonical names (ADDON_NAME in these command help texts).
-  They also application-specific names called aliases or attachments. In
-  many cases, these names can be used interchangeably when unambiguous.
-
-  For example, given a fictional \`slowdb\` add-on named \`slowdb-cubed-1531\`
-  attached as \`FOO_DB\` to \`app1\` and BAR_DB to \`app2\`, the following
-  invocations are considered equivalent:
-
-  $ heroku addons:upgrade slowdb             slowdb:premium
-  $ heroku addons:upgrade --app app1 FOO_DB  slowdb:premium
-  $ heroku addons:upgrade app1::FOO_DB       slowdb:premium
-  $ heroku addons:upgrade app2::BAR_DB       slowdb:premium
-  $ heroku addons:upgrade acme-inc-datastore slowdb:premium
-
-  If the name used is ambiguous (e.g. if you used \`slowdb\` with more than
-  one add-on of that type installed), the command will error due to
-  ambiguity and a more specific identifier will need to be chosen.
+  $ heroku ${topic} --app acme-inc-www
+
+Overview of Add-ons:
+
+  Add-ons are created with the \`addons:create\` command, providing a reference
+  to an add-on service (such as \`heroku-postgresql\`) or a service and plan
+  (such as \`heroku-postgresql:hobby-dev\`).
+
+  At creation, each add-on is given a globally unique name. In addition, each
+  add-on has at least one attachment alias to each application which uses the
+  add-on. In all cases, the owning application will be attached to the add-on.
+  An attachment alias is unique to its application, and is used as a prefix to
+  any environment variables it exports to the application.
+
+  In this example, a \`heroku-postgresql\` add-on is created and its given name
+  is \`postgresql-deep-6913\` with a default attachment alias of \`DATABASE\`:
+
+    $ heroku addons:create heroku-postgresql --app my-app
+    Creating postgresql-deep-6913... done, (free)
+    Adding postgresql-deep-6913 to my-app... done
+    Setting DATABASE_URL and restarting my-app... done, v5
+    Database has been created and is available
+
+    $ heroku addons --app my-app
+    Add-on                                     Plan       Price
+    ─────────────────────────────────────────  ─────────  ─────
+    heroku-postgresql (postgresql-deep-6913)   hobby-dev  free
+    └─ as HEROKU_POSTGRESQL_AMBER
```

Comment:
When I ran through this scenario, I ended up with just the single attachment having a `DATABASE` alias. The output then was:

```
$ heroku addons --app conway-test
Add-on                                               Plan       Price
───────────────────────────────────────────────────  ─────────  ─────
heroku-postgresql (postgresql-vertical-5832)         hobby-dev  free 
 └─ as DATABASE
```